File: LeetCode-DSA/AprilChallenge_2023/AddDigits.py
'''
Q258. Description: Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.
'''

class Solution:
    def addDigits(self, num: int) -> int:
        # Add digits in current number
        def Sum(n):
            single = 0
            while(n):
                dig = n%10
                single += dig
                n = n//10
            return single
        
        # Keep adding digits in number till single digit obtained
        ans = Sum(num)
        while(ans > 9):
            ans = Sum(ans)
        return ans
            

File: LeetCode-DSA/AprilChallenge_2023/AddDigits.cpp
/*
Q258. Description: Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.
*/

class Solution {
public:
    int Sum(int n)
    {
        int total = 0;
        while(n > 0)
        {
            int digit = n % 10;
            total += digit;
            n = n / 10;
        }
        return total;
    }
    
    int addDigits(int num) {
        int res = Sum(num);
        while(res > 9)
            res = Sum(res);
        return res;
    }
};